luarules/testing/util.lua

```diff
@@ -0,0 +1,360 @@
+local serpent = serpent or VFS.Include('common/luaUtilities/serpent.lua')
+
```

Comment:
this file has most of the random functions I could split off from the main widget/gadget files. some could be split into further small files, and some could probably be integrated into existing util files.